services/director-v2/src/simcore_service_director_v2/cli.py

```diff
@@ -0,0 +1,131 @@
+
+        DirectorV0Client.create(
+            app,
+            client=httpx.AsyncClient(
+                base_url=settings.DIRECTOR_V0.endpoint,
+                timeout=app.state.settings.CLIENT_REQUEST.HTTP_CLIENT_REQUEST_TOTAL_TIMEOUT,
+            ),
+        )
+
+        app.state.long_running_client_configuration = ClientConfiguration(
+            router_prefix="", default_timeout=15
+        )
+
+        yield app
+
+    finally:
+        await db.events.close_db_connection(app)
+        await api_client.shutdown(app)
+        client = DirectorV0Client.instance(app).client
+        await client.aclose()
+
+
+def _get_dynamic_sidecar_endpoint(
+    settings: AppSettings, node_id: NodeIDStr
+) -> AnyHttpUrl:
+    dynamic_sidecar_names = DynamicSidecarNames.make(UUID(node_id))
+    hostname = dynamic_sidecar_names.service_name_dynamic_sidecar
+    port = settings.DYNAMIC_SERVICES.DYNAMIC_SIDECAR.DYNAMIC_SIDECAR_PORT
+    return parse_obj_as(AnyHttpUrl, f"http://{hostname}:{port}")  # NOSONAR
+
+
+@main.command()
+def project_save_state(
+    project_id: ProjectID, retry_save: int = DEFAULT_NODE_SAVE_RETRY
+):
+    """
+    Saves the state of all dy-sidecars in a project.
+    In case of error while saving the state of an individual node,
+    it will retry to save.
+    """
+
+    async def _run() -> None:
+        settings = AppSettings.create_from_envs()
+        async with _initialized_app(settings) as app:
+            projects_repository: ProjectsRepository = fetch_repo_outside_of_request(
+                app, ProjectsRepository
+            )
+            project_at_db = await projects_repository.get_project(project_id)
+
+            typer.echo(
+                f"Saving project '{project_at_db.uuid}' - '{project_at_db.name}'"
+            )
+
+            dynamic_sidecar_client = api_client.get_dynamic_sidecar_client(app)
+            for node_uuid, node_content in project_at_db.workbench.items():
+                # onl dynamic-sidecars are used
+                if not await requires_dynamic_sidecar(
+                    service_key=node_content.key,
+                    service_version=node_content.version,
+                    director_v0_client=DirectorV0Client.instance(app),
+                ):
+                    continue
+
+                typer.echo(f"Saving state for {node_uuid} {node_content.label}")
+
+                async for attempt in AsyncRetrying(
+                    wait=wait_random_exponential(),
+                    stop=stop_after_attempt(retry_save),
+                    reraise=True,
+                ):
+                    with attempt:
+                        typer.echo(
+                            f"Attempting to save {node_uuid} {node_content.label}"
+                        )
+                        await dynamic_sidecar_client.save_service_state(
+                            _get_dynamic_sidecar_endpoint(settings, node_uuid)
+                        )
+
+        typer.echo("Save complete")
```

Comment:
what about giving a summary at the end of what was saved and was not saved and requires further manual work?

Reply:
and is there no possibility to re-use the task from the scheduler? that would also prevent code from going out of sync...

Reply:
> what about giving a summary at the end of what was saved and was not saved and requires further manual work?

I've added a summary of the nodes which failed to save (the only interesting one)
Please check the description of the PR



> and is there no possibility to re-use the task from the scheduler? that would also prevent code from going out of sync...

Well `dynamic_sidecar_client.save_service_state` is already using the same code as the scheduler.